py_irt/cli.py

```diff
@@ -36,9 +40,123 @@ def train(
 
 
 @app.command()
-def evaluate():
-    raise NotImplementedError()
+def train_and_evaluate(
+    model_type: str, 
+    data_path: str, 
+    output_dir: str, 
+    epochs: int = 2000, 
+    device: str = "cpu",
+    initializers: Optional[List[str]] = None,
+    evaluation: str = "heldout",
+    seed: int = 42,
+    train_size: float = 0.9,
+):
 
+    console.log(f"model_type: {model_type} data_path: {data_path}")
+    start_time = time.time()
+    config = IrtConfig(model_type=model_type, epochs=epochs, initializers=initializers)
+    if evaluation == "heldout":
+        with open(data_path) as f:
+            items = []
+            for line in f:
+                submission = json.loads(line)
+                model_id = submission["subject_id"]
+                for example_id in submission["responses"].keys():
+                    items.append((model_id, example_id))
+            random.seed(seed)
+            train, test = train_test_split(items, train_size=train_size)
+            training_dict = {}
+            for model_id, example_id in train:
+                training_dict.setdefault(model_id, dict())
+                training_dict[model_id][example_id] = True 
+            for model_id, example_id in test:
+                training_dict.setdefault(model_id, dict())
+                training_dict[model_id][example_id] = False 
+        dataset = Dataset.from_jsonlines(data_path, train_items=training_dict)
+    else:
+        dataset = Dataset.from_jsonlines(data_path)
+
+    # deep copy for training
+    training_data = copy.deepcopy(dataset) 
+    trainer = IrtModelTrainer(config=config, dataset=training_data, data_path=data_path)
+    output_dir = Path(output_dir)
+    console.log("Training Model...")
+    trainer.train(device=device)
+    trainer.save(output_dir / "parameters.json")
+    write_json(output_dir / "best_parameters.json", trainer.best_params)
+
+    # get validation data
+    # filter out test data
+    console.log("Evaluating Model...")
+    testing_idx = [i for i in range(len(dataset.training_example)) if not dataset.training_example[i]]
+    if len(testing_idx) > 0:
+        dataset.observation_subjects = [dataset.observation_subjects[i] for i in testing_idx]
+        dataset.observation_items= [dataset.observation_items[i] for i in testing_idx]
+        dataset.observations = [dataset.observations[i] for i in testing_idx]
+        dataset.training_example = [dataset.training_example[i] for i in testing_idx]
+
+    preds = trainer._irt_model.predict(dataset.observation_subjects, dataset.observation_items)
```

Comment:
We should probably make the model `trainer.irt_model` instead of `trainer._irt_model` if we're calling it here

Reply:
Agree, this has been updated throughout.